### PR TITLE
Move GCP functions from WorkspaceManagerService.java

### PR DESCRIPTION
**Pull request by dexamundsen**
- Split WorkspaceManagerService.java --> this util file included general workspace ops, utils and GCP ops calls to WSM. GCP calls is moved to a separate file WorkspaceManagerServiceGCP.java
- This change helps improve code readability and maintenance when other cloud functions are added
- Use strongly types variables - reason: better readability, dev time type inference w/o depending on compiler
- No functional changes

### CODE REVIEW

File: src/main/java/bio/terra/cli/command/user/sshkey/Add.java [hunk 1]
```diff
@@ -25,15 +25,16 @@ public class Add extends BaseCommand {
...
     ExternalCredentialsManagerService ecmService = ExternalCredentialsManagerService.fromContext();
-    var sshKeyPair = ecmService.getSshKeyPair(SshKeyPairType.GITHUB, /*includePrivateKey=*/ true);
+    SshKeyPair sshKeyPair =
+        ecmService.getSshKeyPair(SshKeyPairType.GITHUB, /*includePrivateKey=*/ true);
     saveKeyFileAndSshAdd(sshKeyPair);
     formatOption.printReturnValue(UFSshKeyPair.createUFSshKey(sshKeyPair), UFSshKeyPair::print);
   }
 
   protected static void saveKeyFileAndSshAdd(SshKeyPair sshKeyPair) {
-    var dir = System.getProperty("user.home") + "/.ssh/";
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/command/user/sshkey/Add.java:34]: i still think we don't need to specify they are string, it's pretty clear

File: src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java [hunk 2]
```diff
@@ -40,8 +41,8 @@ public static boolean isOlder(
       return false;
     }
 
-    var currentVersion = Version.parse(currentVersionString);
-    var oldestSupportedVersion = Version.parse(oldestSupportedVersionString);
+    ModuleDescriptor.Version currentVersion = Version.parse(currentVersionString);
```
Comment (zloery) [inline src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java:44]: In general, I don't think we need to get rid of `var` when it's clear what the object type is e.g. `Version.parse` returning a `Version` object

File: src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java [hunk 3]
```diff
@@ -0,0 +1,747 @@
...
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility methods for calling Workspace Manager endpoints. */
```
Comment (zloery) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java:76]: nit: specifically for calling WSM's GCP endpoints

**Review by zloery — APPROVED** [review]

File: src/main/java/bio/terra/cli/command/user/sshkey/Add.java (hunk 1 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/command/user/sshkey/Add.java:34]: reverting string change here. Not against string being inferred

File: src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import bio.terra.cli.businessobject.Context;
 import bio.terra.cli.service.WorkspaceManagerService;
 import bio.terra.workspace.model.SystemVersion;
+import java.lang.module.ModuleDescriptor;
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java:6]: it is already included, removing duplicate

File: src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java (hunk 3 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java:76]: done

File: src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java (hunk 2 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java:44]: noted

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**